logger_cb.go

```diff
@@ -0,0 +1,157 @@
+package libbpfgo
+
+/*
+#include <bpf/libbpf.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+)
+
+// This callback definition needs to be in a different file from where it is declared in C
+// Otherwise, multiple definition compilation error will occur
+
+// loggerCallback is called by libbpf_print_fn() which in turn is called by libbpf
+//
+//export loggerCallback
+func loggerCallback(libbpfPrintLevel int, libbpfOutput *C.char) {
+	var (
+		level    int
+		goOutput string
+	)
+
+	goOutput = C.GoString(libbpfOutput)
+	goOutput = strings.TrimSuffix(goOutput, "\n")
+
+	for _, fnFilterOut := range logFnFilters {
+		if fnFilterOut != nil {
+			if fnFilterOut(libbpfPrintLevel, goOutput) {
+				return
+			}
+		}
+	}
+
+	log(level, goOutput)
+}
+
+const (
+	LogWarnLevel  = int(C.LIBBPF_WARN)
+	LogInfoLevel  = int(C.LIBBPF_INFO)
+	LogDebugLevel = int(C.LIBBPF_DEBUG)
+)
+
+var log = logFallback
+var logFnFilters = []func(libLevel int, msg string) bool{
+	LogFilterLevel,
+	LogFilterOutput,
+}
+
+// SetLoggerCbs receives a function to be used to log libbpf outputs and
+// a list of functions to filter this output out
+// If fnLog is nil, logFallback will continue to be used
+// If fnFilters is nil, logFnFilters will continue to be used
+func SetLoggerCbs(
+	fnLog func(level int, msg string, keyValues ...interface{}), // log callback
```

Comment:
Both of these can be in a struct, then passing nil for both would look like
`libbpfgo.SetLoggerCbs(SettingStructName{})` and not `libbpfgo.SetLoggerCbs(nil, nil)` which I think is easier to understand (empty struct => default setting compared to passing NULL)

Reply:
> I had to look at the equivalent tracee PR to understand the first setting, and I have no idea how the filtering would work, maybe tests could help here.

I agree. A test example is welcome. Will work on that.

> Both of these can be in a struct, then passing nil for both would look like
libbpfgo.SetLoggerCbs(SettingStructName{}) and not libbpfgo.SetLoggerCbs(nil, nil) which I think is easier to understand (empty struct => default setting compared to passing NULL)

It makes sense. 👍🏼